### PR TITLE
Set OPENSSL_ROOT_DIR for cmake if not set

### PR DESCRIPTION
**Pull request by bernedom**
If the CMake files are included they set the OPENSSL_ROOT_DIR variable for CMake as described in the documentation:
https://cmake.org/cmake/help/latest/module/FindOpenSSL.html 

All libraries are symlinked to open to a version without the version (i.e libcrypto.so -> libcrypto_1.1.so)

This will allow people to use `find_package(OpenSSL)` in their root CMakeLists.txt after including the cmake files.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+  if(NOT OPENSSL_ROOT_DIR)
+    set(OPENSSL_ROOT_DIR ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI})
+  endif()
+  list(APPEND ANDROID_EXTRA_LIBS
+       ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libcrypto_1_1.so
+       ${SSL_ROOT_PATH}/ssl_1.1/${CMAKE_ANDROID_ARCH_ABI}/libssl_1_1.so)
 endif()
-
-set_target_properties(${PROJECT_NAME} PROPERTIES
-    QT_ANDROID_EXTRA_LIBS "${ANDROID_EXTRA_LIBS}")
```
Comment (bog-dan-ro) [inline CMakeLists.txt:18]: I'm not a cmake expert, but I think the `QT_ANDROID_EXTRA_LIBS` is still needed.

**Review by bog-dan-ro — COMMENTED** [review]
@bernedom  Thanks for your patches!

I don't know how the symlinks will work on windows,  let's wait a bit also for @Issam-b's opinion on this.

File: CMakeLists.txt (hunk 1 above)
Comment (bernedom) [reply CMakeLists.txt:18]: It is not needed in all cases. Also this fails if there is no target named after `${PROJECT_NAME}` which is not always the case. 

Some consider it actually bad practice to reuse the `${PROJECT_NAME}` for target name, even though it is used frequently. 
As there is also the `android_openssl.cmake` file which has a function to assign the libs to the target properties.

**Comment by bernedom** [comment]
@bog-dan-ro @Issam-b if the symlinks do not work on windows my guess is that the resolution with `find_package(OpenSSL)` also does not work on windows. I unfortunately cannot test this as I have no windows machine around.

**Pull request merged**